### PR TITLE
Refactor Quota Source method and spec

### PR DESCRIPTION
**Pull request by GregP**
This PR is based on the issue below:
ManageIQ/manageiq#12038

@miq-bot add_label refactoring

### CODE REVIEW

File: content/automate/ManageIQ/System/CommonMethods/QuotaMethods.class/__methods__/quota_source.rb [hunk 1]
```diff
@@ -1,16 +1,43 @@
+# frozen_string_literal: true
```
Comment (tinaafitz) [inline content/automate/ManageIQ/System/CommonMethods/QuotaMethods.class/__methods__/quota_source.rb]: Hi @GregP I don't see this line in the other methods. Does it resolve a rubocop error?

File: content/automate/ManageIQ/System/CommonMethods/QuotaMethods.class/__methods__/quota_source.rb [hunk 2]
```diff
@@ -1,16 +1,43 @@
...
-when 'group'
-  $evm.root['quota_source'] = @miq_request.requester.current_group
-when 'user'
-  $evm.root['quota_source'] = @miq_request.requester
-else
-  $evm.root['quota_source'] = @miq_request.tenant
-  $evm.root['quota_source_type'] = 'tenant'
+            def quota_source
+              @miq_request = @handle.root['miq_request']
+              # @handle.root['quota_source_type'] =  @handle.object['quota_source_type']
```
Comment (tinaafitz) [inline content/automate/ManageIQ/System/CommonMethods/QuotaMethods.class/__methods__/quota_source.rb]: Can we remove this commented out line?

File: content/automate/ManageIQ/System/CommonMethods/QuotaMethods.class/__methods__/quota_source.rb (hunk 1 above)
Comment (GregP) [reply content/automate/ManageIQ/System/CommonMethods/QuotaMethods.class/__methods__/quota_source.rb]: @tinaafitz This line should be there, speeds up processing until we move to a more recent Ruby release.

File: content/automate/ManageIQ/System/CommonMethods/QuotaMethods.class/__methods__/quota_source.rb [hunk 3]
```diff
@@ -1,16 +1,43 @@
...
-case $evm.root['quota_source_type'].downcase
-when 'group'
-  $evm.root['quota_source'] = @miq_request.requester.current_group
-when 'user'
-  $evm.root['quota_source'] = @miq_request.requester
-else
-  $evm.root['quota_source'] = @miq_request.tenant
-  $evm.root['quota_source_type'] = 'tenant'
+            def quota_source
+              @miq_request = @handle.root['miq_request']
```
Comment (gmcculloug) [inline content/automate/ManageIQ/System/CommonMethods/QuotaMethods.class/__methods__/quota_source.rb]: There is no reason to for `@miq_request` to be an instance variable.  Also, part of breaking these methods down is to separate out some of the logic into methods.

I would suggest getting rid of the `@miq_request` variable all together and creating a `miq_request` method that just gets called when needed:
```ruby
def miq_request
  @handle.root['miq_request']
end
```

Then all the callers change to use the method.  For example:
```diff
- @handle.root['quota_source'] = @miq_request.requester.current_group
+ @handle.root['quota_source'] = miq_request.requester.current_group

```

File: content/automate/ManageIQ/System/CommonMethods/QuotaMethods.class/__methods__/quota_source.rb (hunk 2 above)
Comment (GregP) [reply content/automate/ManageIQ/System/CommonMethods/QuotaMethods.class/__methods__/quota_source.rb]: @tinaafitz Done.

File: content/automate/ManageIQ/System/CommonMethods/QuotaMethods.class/__methods__/quota_source.rb (hunk 3 above)
Comment (GregP) [reply content/automate/ManageIQ/System/CommonMethods/QuotaMethods.class/__methods__/quota_source.rb]: Code refactored.

File: content/automate/ManageIQ/System/CommonMethods/QuotaMethods.class/__methods__/quota_source.rb [hunk 4]
```diff
@@ -1,16 +1,45 @@
...
+  module Automate
+    module System
+      module CommonMethods
+        module QuotaMethods
+          class QuotaSource
+            def initialize(handle = $evm)
+              @handle = handle
+            end
+
+            def miq_request
```
Comment (gmcculloug) [inline content/automate/ManageIQ/System/CommonMethods/QuotaMethods.class/__methods__/quota_source.rb]: @GregP This method belongs in the `private` method section.

**Pull request merged**